### PR TITLE
Migrate to Gradle version catalogs

### PR DESCRIPTION
**Pull request by Maxr1998**
I guess since everything builds, it shouldn't require a very close review.. but it won't hurt either.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -19,13 +19,13 @@ allprojects {
         google()
         mavenLocal {
             content {
-                includeVersionByRegex(Dependencies.Groups.jellyfin, ".*", Dependencies.Versions.jellyfinSdkLocal)
+                includeVersionByRegex(JellyfinSdk.GROUP, ".*", JellyfinSdk.LOCAL)
```
Comment (nielsvanvelzen) [inline build.gradle.kts:22]: Why not `libs.versions.jellyfinSdkLocal`?

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+androidx-test-runner = "1.4.0"
+androidx-test-espresso = "3.4.0"
+
+[libraries]
+# KotlinX
+coroutines-core = { group = "org.jetbrains.kotlinx", name = "kotlinx-coroutines-core", version.ref = "coroutines" }
+coroutines-android = { group = "org.jetbrains.kotlinx", name = "kotlinx-coroutines-android", version.ref = "coroutines" }
+
+# Core
+koin = { group = "io.insert-koin", name = "koin-android", version.ref = "koin" }
```
Comment (nielsvanvelzen) [inline gradle/libs.versions.toml:57]: I don;t see a point in separating the version if it's only used once.

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+# Testing
+junit = "5.7.2"
+kotest = "4.6.1"
+mockk = "1.12.0"
+androidx-test-runner = "1.4.0"
+androidx-test-espresso = "3.4.0"
+
+[libraries]
+# KotlinX
+coroutines-core = { group = "org.jetbrains.kotlinx", name = "kotlinx-coroutines-core", version.ref = "coroutines" }
```
Comment (nielsvanvelzen) [inline gradle/libs.versions.toml:53]: What are your rules to determine between group+name or module?

Comment (nielsvanvelzen) [reply gradle/libs.versions.toml:53]: (IMO module is always shorter and therefor easier to read)

File: gradle/libs.versions.toml [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+coroutines-android = { group = "org.jetbrains.kotlinx", name = "kotlinx-coroutines-android", version.ref = "coroutines" }
+
+# Core
+koin = { group = "io.insert-koin", name = "koin-android", version.ref = "koin" }
+androidx-core = { module = "androidx.core:core-ktx", version.ref = "androidx-core" }
+androidx-appCompat = { module = "androidx.appcompat:appcompat", version.ref = "androidx-appcompat" }
```
Comment (nielsvanvelzen) [inline gradle/libs.versions.toml:59]:
```suggestion
androidx-appcompat = { module = "androidx.appcompat:appcompat", version.ref = "androidx-appcompat" }
```

Officially called "Appcompat" without uppercase C

File: gradle/libs.versions.toml [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+androidx-activity = { module = "androidx.activity:activity-ktx", version.ref = "androidx-activity" }
+androidx-fragment = { module = "androidx.fragment:fragment-ktx", version.ref = "androidx-fragment" }
+androidDesugarLibs = { module = "com.android.tools:desugar_jdk_libs", version.ref = "androidDesugarLibs" }
+
+# Lifecycle Extensions
+androidx-lifecycleExtensions-viewModel = { module = "androidx.lifecycle:lifecycle-viewmodel-ktx", version.ref = "androidx-lifecycleExtensions" }
```
Comment (nielsvanvelzen) [inline gradle/libs.versions.toml:65]: I'd shorten it to `androidx-lifecycle-xxx`, also viewmodel is without uppercase M. https://developer.android.com/jetpack/androidx/releases/lifecycle

File: gradle/libs.versions.toml [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+androidx-lifecycleExtensions-liveData = { module = "androidx.lifecycle:lifecycle-livedata-ktx", version.ref = "androidx-lifecycleExtensions" }
```
Comment (nielsvanvelzen) [inline gradle/libs.versions.toml:66]: no uppercase D, actually should just lowercase everything

File: buildSrc/src/main/kotlin/JellyfinSdk.kt [hunk 7]
```diff
@@ -0,0 +1,7 @@
+object JellyfinSdk {
+    const val GROUP ="org.jellyfin.sdk"
```
Comment (nielsvanvelzen) [inline buildSrc/src/main/kotlin/JellyfinSdk.kt:2]:
```suggestion
    const val GROUP = "org.jellyfin.sdk"
```

**Review by nielsvanvelzen — CHANGES_REQUESTED** [review]

File: build.gradle.kts (hunk 1 above)
Comment (Maxr1998) [reply build.gradle.kts:22]: Good point, I'll fix that. Forgot that this worked too.

File: gradle/libs.versions.toml (hunk 2 above)
Comment (Maxr1998) [reply gradle/libs.versions.toml:57]: I disagree, it allows you to see all versions in a single list.

File: gradle/libs.versions.toml (hunk 3 above)
Comment (Maxr1998) [reply gradle/libs.versions.toml:53]: It was a bit random, I mostly split group/name when having multiple of the same group. But it's not really consistent. Not sure what'd be the recommended style (shorter is not really an argument imo).

File: gradle/libs.versions.toml (hunk 4 above)
Comment (Maxr1998) [reply gradle/libs.versions.toml:59]: Right, that's a copy paste error. I fixed it for some other deps.

File: buildSrc/src/main/kotlin/JellyfinSdk.kt (hunk 7 above)
Comment (Maxr1998) [reply buildSrc/src/main/kotlin/JellyfinSdk.kt:2]: Huh, how did that happen..

File: build.gradle.kts (hunk 1 above)
Comment (Maxr1998) [reply build.gradle.kts:22]: Nvm, that doesn't work. It fails the build.

**Comment by Maxr1998** [comment]
I recommend a squash after review ^^

File: build.gradle.kts (hunk 1 above)
Comment (nielsvanvelzen) [reply build.gradle.kts:22]: Maybe it needs Gradle 7.2 :(

Comment (Maxr1998) [reply build.gradle.kts:22]: I'd merge it as is then.

**Review by nielsvanvelzen — APPROVED** [review]

**Pull request merged**